### PR TITLE
Removing '\n' from the logs of zedagent

### PR DESCRIPTION
**Pull request by zed-rishabh**
Signed-off-by: zed-rishabh <rgupta@zededa.com>

### CODE REVIEW

File: pkg/pillar/cmd/zedagent/handlecertconfig.go [hunk 1]
```diff
@@ -37,10 +37,10 @@ func parseControllerCerts(ctx *zedagentContext, contents []byte) {
...
 		return
 	}
-	log.Infof("parseControllerCerts: Applying updated config\n"+
-		"Last Sha: % x\n"+
-		"New  Sha: % x\n"+
-		"Num of cfgCert: %d\n",
+	log.Infof("parseControllerCerts: Applying updated config"+
+		"Last Sha: % x"+
+		"New  Sha: % x"+
+		"Num of cfgCert: %d",
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/handlecertconfig.go:43]: I think this was intended to be human readable with line breaks. But perhaps that doesn't make sense with json.
But in any case, you need to replace the intermediate \n with a space or something - right now you are concantating  words and sha values.

File: pkg/pillar/cmd/zedagent/handlecipherconfig.go [hunk 2]
```diff
@@ -31,10 +31,10 @@ func parseCipherContext(ctx *getconfigContext,
...
 		return
 	}
-	log.Infof("parseCipherContext: Applying updated config\n"+
-		"Last Sha: % x\n"+
-		"New  Sha: % x\n"+
-		"Num of cfgCipherContext: %d\n",
+	log.Infof("parseCipherContext: Applying updated config"+
+		"Last Sha: % x"+
+		"New  Sha: % x"+
+		"Num of cfgCipherContext: %d",
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/handlecipherconfig.go:37]: Ditto

File: pkg/pillar/cmd/zedagent/parseconfig.go [hunk 3]
```diff
@@ -109,10 +109,10 @@ func parseBaseOsConfig(getconfigCtx *getconfigContext,
...
 		return
 	}
-	log.Infof("parseBaseOsConfig: Applying updated config\n"+
-		"prevSha: % x\n"+
-		"NewSha : % x\n"+
-		"cfgOsList: %v\n",
+	log.Infof("parseBaseOsConfig: Applying updated config"+
+		"prevSha: % x"+
+		"NewSha : % x"+
+		"cfgOsList: %v",
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/parseconfig.go:115]: Ditto

File: pkg/pillar/cmd/zedagent/parseconfig.go [hunk 4]
```diff
@@ -193,10 +193,10 @@ func parseNetworkXObjectConfig(config *zconfig.EdgeDevConfig,
...
 		return false
 	}
-	log.Infof("parseNetworkXObjectConfig: Applying updated config.\n"+
-		"prevSha: % x\n"+
-		"NewSha : % x\n"+
-		"networks: %v\n",
+	log.Infof("parseNetworkXObjectConfig: Applying updated config."+
+		"prevSha: % x"+
+		"NewSha : % x"+
+		"networks: %v",
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/parseconfig.go:199]: Ditto

File: pkg/pillar/cmd/zedagent/parseconfig.go [hunk 5]
```diff
@@ -217,13 +217,13 @@ func unpublishDeletedNetworkInstanceConfig(ctx *getconfigContext,
...
 			// Entry not deleted.
-			log.Infof("NetworkInstance %s (Name: %s) still exists\n",
+			log.Infof("NetworkInstance %s (Name: %s) still exists",
 				key, networkInstanceEntry.Displayname)
 			continue
 		}
 
 		config := entry.(types.NetworkInstanceConfig)
-		log.Infof("unpublishing NetworkInstance %s (Name: %s) \n",
+		log.Infof("unpublishing NetworkInstance %s (Name: %s) ",
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/parseconfig.go:226]: Would be good to remove the trailing space as well

File: pkg/pillar/cmd/zedagent/parseconfig.go [hunk 6]
```diff
@@ -433,10 +433,10 @@ func parseNetworkInstanceConfig(config *zconfig.EdgeDevConfig,
...
 		return
 	}
-	log.Infof("parseNetworkInstanceConfig: Applying updated config\n"+
-		"prevSha: % x\n"+
-		"NewSha : % x\n"+
-		"networkInstances: %v\n",
+	log.Infof("parseNetworkInstanceConfig: Applying updated config"+
+		"prevSha: % x"+
+		"NewSha : % x"+
+		"networkInstances: %v",
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/parseconfig.go:439]: Ditto

File: pkg/pillar/cmd/zedagent/parseconfig.go [hunk 7]
```diff
@@ -458,10 +458,10 @@ func parseAppInstanceConfig(config *zconfig.EdgeDevConfig,
...
 		return
 	}
-	log.Infof("parseAppInstanceConfig: Applying updated config\n"+
-		"prevSha: % x\n"+
-		"NewSha : % x\n"+
-		"Apps: %v\n",
+	log.Infof("parseAppInstanceConfig: Applying updated config"+
+		"prevSha: % x"+
+		"NewSha : % x"+
+		"Apps: %v",
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/parseconfig.go:464]: Ditto

File: pkg/pillar/cmd/zedagent/parseconfig.go [hunk 8]
```diff
@@ -584,11 +584,11 @@ func parseSystemAdapterConfig(config *zconfig.EdgeDevConfig,
...
-		log.Infof("parseSystemAdapterConfig: Applying updated config\n"+
-			"prevSha: % x\n"+
-			"NewSha : % x\n"+
-			"sysAdapters: %v\n"+
-			"Forced parsing: %v\n",
+		log.Infof("parseSystemAdapterConfig: Applying updated config"+
+			"prevSha: % x"+
+			"NewSha : % x"+
+			"sysAdapters: %v"+
+			"Forced parsing: %v",
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/parseconfig.go:591]: Ditto

**Comment by rvs** [comment]
Hey @zed-rishabh  -- do you plan to get this fix soon? Just doing some spring cleaning on our PRs and came across this one.

**Comment by zed-rishabh** [comment]
> Hey @zed-rishabh -- do you plan to get this fix soon? Just doing some spring cleaning on our PRs and came across this one.

Hey @rvs, it will get fix today.

File: pkg/pillar/cmd/zedagent/handlecertconfig.go (hunk 1 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/zedagent/handlecertconfig.go]: Resolved

File: pkg/pillar/cmd/zedagent/handlecipherconfig.go (hunk 2 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/zedagent/handlecipherconfig.go]: Resolved

File: pkg/pillar/cmd/zedagent/parseconfig.go (hunk 4 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/zedagent/parseconfig.go]: Resolved

File: pkg/pillar/cmd/zedagent/parseconfig.go (hunk 3 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/zedagent/parseconfig.go]: Resolved

File: pkg/pillar/cmd/zedagent/parseconfig.go (hunk 5 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/zedagent/parseconfig.go]: Resolved

File: pkg/pillar/cmd/zedagent/parseconfig.go (hunk 6 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/zedagent/parseconfig.go]: Resolved

File: pkg/pillar/cmd/zedagent/parseconfig.go (hunk 7 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/zedagent/parseconfig.go]: Resolved

File: pkg/pillar/cmd/zedagent/parseconfig.go (hunk 8 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/zedagent/parseconfig.go]: Resolved

**Comment by zed-rishabh** [comment]
> One case of a trailing space (see comments) and a bunch of cases (didn't mark all of them) where the code prints prevdsha/newsha over several lines.
> In actually don't think we need to print the sha values for any of those; was useful when initially debugging the code years ago.
> 
> Also, printing with %v in those might create a message which is too larger, but that is a separate issue.

I will take it separately of removing SHA from the log messages.

**Pull request merged**